[Workflows][Connectors] Add Ansible Control Server connector for AWX/Controller jobs (#280115)
elastic/kibana: docs/reference/connectors-kibana/ansible-controller-action-type.md

---
navigation_title: "Ansible Control Server"
type: reference
description: "Use the Ansible Control Server connector to launch and monitor Ansible Automation Controller / AWX jobs."
applies_to:

>   stack: preview 9.4

Reviewer: ```suggestion
  stack: preview 9.6
```

:   The base URL of the Controller / AWX server, for example `https://controller.example.com`. This host must be permitted by the [`xpack.actions.allowedHosts`](/reference/configuration-reference/alerting-settings.md#action-settings) setting.

API base path
:   API root path prepended to typed actions. Use `/api/v2` for AWX, Tower, and AAP ≤2.4. Use `/api/controller/v2` for AAP 2.5+ platform gateway installs. Default: `/api/v2`.

### Authentication [ansible-controller-connector-authentication]

**API token**

Token
:   A long-lived personal access token (PAT) or OAuth2 application token. The connector sends it in the `Authorization: Bearer <token>` header. Do **not** paste short-lived session cookies.

Server CA certificate (PEM)
:   Optional PEM-encoded certificate authority used to verify the server certificate.

Verification mode
:   How to verify the server TLS certificate: `full` (default), `certificate`, or `none` (not recommended).

## Test connectors [ansible-controller-action-configuration]

You can test connectors when you create or edit the connector in {{kib}}. The test requests the authenticated user (`GET …/me/`) to verify connectivity and authentication.

> 

Reviewer: ```suggestion

## Connection actions [ansible-controller-connector-actions]

```
Likely best to list the actions in a clear section than under "Test"

The connector has the following actions:

`request`
:   Make an authenticated request to any Controller / AWX API path. Prefer typed actions when they fit.

`listJobTemplates` / `getJobTemplate` / `getJobTemplateLaunchOptions`
:   Discover templates and launch-time prompts (`ask_*`, survey defaults).

`launchJobTemplate`

> :   Launch a job (`extra_vars`, `limit`, `inventory`, credentials, tags, etc.). Mutates infrastructure.

Reviewer: ```suggestion
:   Launch a job (`extra_vars`, `limit`, `inventory`, credentials, tags, and so on). Mutates infrastructure.
```
